### PR TITLE
Documented estimates

### PR DESCRIPTION
**Pull request by g4s8**
#1236 - estimates documentation

### CODE REVIEW

**Comment by g4s8** [comment]
@0crat assign @amihaiemil

**Comment by 0crat** [comment]
Job [#1252](https://github.com/zerocracy/farm/issues/1252) is now in scope, role is `REV`

**Comment by 0crat** [comment]
> @0crat assign @amihaiemil  ([here](https://github.com/zerocracy/farm/issues/1252#issuecomment-400246527))

@g4s8 This pull request [#1252](https://github.com/zerocracy/farm/issues/1252) is assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/1770950729); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by 0crat** [comment]
Manual assignment of issues is discouraged, see [§19](http://www.zerocracy.com/policy.html#19): -5 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

File: src/main/java/com/zerocracy/pm/cost/Estimates.java [hunk 1]
```diff
@@ -37,13 +37,32 @@
...
+ * which will be payed for a job on complete.
+ * For example this estimate:
+ * <pre>
+ * <code>&lt;order id="gh:yegor256/pdd#3"&gt;
+ *   &lt;cash&gt;$15&lt;/cash&gt;
+ *   &lt;created&gt;2016-12-29T09:03:21.684Z&lt;/created&gt;
+ *   &lt;role&gt;REV&lt;/role&gt;
+ * &lt;/order&gt;
+ * </code>
+ * </pre>
```
Comment (amihaiemil) [inline src/main/java/com/zerocracy/pm/cost/Estimates.java]: @g4s8 Why is there both ``<code>`` and ``<pre>``? I think ``<pre>`` is enough. And have you checked how the javadoc renders, is it ok? :D

File: src/main/java/com/zerocracy/pm/cost/Estimates.java [hunk 2]
```diff
@@ -37,13 +37,32 @@
...
+ * means that job {@code gh:yegor256/pdd#3} was estimated in $15 as a
+ * code-review task.<br/>
+ * Project estimates locks a cash from budget, so if estimated value is
+ * bigger that project's cash, project will be turned into deficit mode.
+ * <p>
+ * Estimate can be cleaned from {@link com.zerocracy.farm.ruled.RdAuto},
```
Comment (amihaiemil) [inline src/main/java/com/zerocracy/pm/cost/Estimates.java]: @g4s8 If the estimates are for a particular job, why would they ever need to be cleaned/removed? They are calculated at the beginning of a job, the job is eventually finished and payments are done -- why do we care about the estimates anymore and have to remove them?

File: src/main/java/com/zerocracy/pm/cost/Estimates.java [hunk 3]
```diff
@@ -37,13 +37,32 @@
 
 /**
  * Cost estimates.
+ * <p>
```
Comment (amihaiemil) [inline src/main/java/com/zerocracy/pm/cost/Estimates.java]: @g4s8 I think this is not closed properly, there is another ``<p>`` bellow, but I can't see any ``</p>`` anywhere

File: src/main/java/com/zerocracy/pm/cost/Estimates.java (hunk 1 above)
Comment (g4s8) [reply src/main/java/com/zerocracy/pm/cost/Estimates.java]: @amihaiemil yes, I've checked it in Idea. We need `<code>` here for monospace font and `<pre>` to keep indentations in javadocs:
scrot-2018-06-27-11 46 54

File: src/main/java/com/zerocracy/pm/cost/Estimates.java (hunk 2 above)
Comment (g4s8) [reply src/main/java/com/zerocracy/pm/cost/Estimates.java]: @amihaiemil estimates represents only current state of "pending payments", e.g. in this project estimates are `$394.00`, it means that project will pay `$394.00` for current orders on complete. Also it needs to calculate deficit in the project: if project's budget is less than estimates, then it's deficit (can't pay for all current orders). 

Updated javadoc also.

File: src/main/java/com/zerocracy/pm/cost/Estimates.java (hunk 3 above)
Comment (g4s8) [reply src/main/java/com/zerocracy/pm/cost/Estimates.java]: @amihaiemil fixed

**Comment by amihaiemil** [comment]
@g4s8 I see, looks better now. Good to merge :D

**Comment by g4s8** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1252](https://github.com/zerocracy/farm/issues/1252) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by ypshenychka** [comment]
@amihaiemil According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only two issues were found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by amihaiemil** [comment]
@ypshenychka confirmed

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**